lib/TH/generic/THTensorMath.c

```diff
@@ -358,10 +358,27 @@ accreal THTensor_(dot)(THTensor *tensor, THTensor *src)
   return sum;
 }
 
-real THTensor_(minall)(THTensor *tensor)
+void THTensor_(flatIndexToFullIndex)(long ind, long *index_data, long nDim, long *sizes){
+    double total = 1;
```

Comment:
Shouldn't you use `long` instead of `double`? In the first division, `total` is a multiple of `sizes[i]` and in the second division you floor anyway. So I don't think you need double division at all.